### PR TITLE
Added iOS Image Embedder Options, Result, Helpers and Header

### PR DESCRIPTION
**Pull request by priankakariatyml**

### CODE REVIEW

**Comment by priankakariatyml** [comment]
@schmidt-sebastian PR for Image Embedder

File: mediapipe/tasks/ios/vision/image_embedder/sources/MPPImageEmbedder.h [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+/**
+ * Performs embedding extraction on the provided `MPImage` using the whole image as region of
+ * interest. Rotation will be applied according to the `orientation` property of the provided
+ * `MPImage`. Only use this method when the `ImageEmbedder` is created with running mode, `.image`.
+ *
+ * This method supports embedding extraction on RGBA images. If your `MPImage` has a source type of
+ * `.pixelBuffer` or `.sampleBuffer`, the underlying pixel buffer must have one of the following
+ * pixel format types:
+ * 1. kCVPixelFormatType_32BGRA
+ * 2. kCVPixelFormatType_32RGBA
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/vision/image_embedder/sources/MPPImageEmbedder.h:78]: Nit: Need to update to new phrasing.

File: mediapipe/tasks/ios/vision/image_embedder/sources/MPPImageEmbedder.h [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+ *
+ * @return `true` if the image was sent to the task successfully, otherwise `false`.
+ */
+- (BOOL)embedAsyncImage:(MPPImage *)image
+    timestampInMilliseconds:(NSInteger)timestampInMilliseconds
+           regionOfInterest:(CGRect)roi
+                      error:(NSError **)error
+    NS_SWIFT_NAME(embedAsync(image:timestampInMilliseconds:regionOfInterest:));
+
+- (instancetype)init NS_UNAVAILABLE;
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/vision/image_embedder/sources/MPPImageEmbedder.h:258]: We should have a cosine similarity API as well: https://github.com/google/mediapipe/blob/master/mediapipe/tasks/python/vision/image_embedder.py#L339

**Review by schmidt-sebastian — CHANGES_REQUESTED** [review]
Thank you! Can you add the cosine similarity API that we use in Text Embedder?

**Review by schmidt-sebastian — APPROVED** [review]

**Pull request merged**